astropy/visualization/wcsaxes/tests/test_misc.py

```diff
@@ -499,6 +500,37 @@ def test_simplify_labels_usetex(ignore_matplotlibrc, tmp_path):
     fig.savefig(tmp_path / "plot.png")
 
 
+@pytest.mark.parametrize("usetex", [True, False])
+@pytest.mark.parametrize("unicode_minus", [True, False])
+def test_simplify_labels_minus_sign(ignore_matplotlibrc, usetex, unicode_minus):
+    # Ensure minus signs aren't removed from the front of labels across a grid of configuration possibilities
+    if usetex and TEX_UNAVAILABLE:
+        pytest.skip("TeX is unavailable")
+    with mpl.rc_context(
+        rc={"text.usetex": usetex, "axes.unicode_minus": unicode_minus}
+    ):
+        ticklabels = TickLabels(None)
+        expected_labels = []
+        for i in [1, 2]:
+            if usetex:
+                label = f"$-{i}$"
+            elif unicode_minus:
+                label = f"\N{MINUS SIGN}{i}"
+            else:
+                label = f"-{i}"
+            ticklabels.add(
+                axis="axis",
+                world=0,
+                angle=0,
+                text=label,
+                axis_displacement=0,
+                data=(0, 0),
```

Comment:
I know we're not going to *look* at these labels, so overlaps don't really matter, but it's still bugging me. Can we make `data` depend on `i` ?